pkg/test/framework/components/apps/native.go

```diff
@@ -357,27 +375,24 @@ func (a *nativeApp) Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedResp
 		return nil, err
 	}
 
-	if len(resp) != 1 {
-		return nil, fmt.Errorf("unexpected number of responses: %d", len(resp))
-	}
-	if !resp[0].IsOK() {
-		return nil, fmt.Errorf("unexpected response status code: %s", resp[0].Code)
-	}
-	if resp[0].Host != dstServiceName {
-		return nil, fmt.Errorf("unexpected host: %s", resp[0].Host)
-	}
-	if resp[0].Port != strconv.Itoa(dst.port.ApplicationPort) {
-		return nil, fmt.Errorf("unexpected port: %s", resp[0].Port)
-	}
-
 	return resp, nil
 }
 
-func (a *nativeApp) CallOrFail(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse {
+func (a *nativeApp) ValidatedCall(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse {
 	r, err := a.Call(e, opts)
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	if len(r) != 1 {
+		t.Fatalf("unexpected number of responses: %d", len(r))
+	}
+	if !r[0].IsOK() {
+		t.Fatalf("unexpected response status code: %s", r[0].Code)
+	}
+	if r[0].Host != e.(*endpoint).owner.Name() {
```

Comment:
Why removed `opts.Headers.Get("Host") != ""` check from the previous function, i.e. `Call()`?

Reply:
I'm still confused by this @howardjohn. Previously, inside `Call` we check
`if resp[0].Host != dstServiceName` where `dstServiceName` could be `e.(*endpoint).owner.Name()` or `opts.Headers.Get("Host")`. Is it safe to ignore `opts.Headers.Get("Host")`?

Reply:
Good catch! the header check got added between writing this and updating this so I didn't notice